graph.cpp

```diff
@@ -65,7 +66,7 @@ void Graph::batch_update(uint64_t src, const std::vector<uint64_t>& edges) {
 
 vector<set<Node>> Graph::connected_components() {
   bf->force_flush(); // flush everything in buffertree to make final updates
-  GraphWorker::stopWorkers(); // tell the workers to stop and wait for them to finish
+  GraphWorker::pause_workers(); // wait for the workers to finish applying the updates
```

Comment:
Below this code there is a variable we use to disallow all future insertions to the graph after running the connected components. Will this be a problem or does code in some other branch address this

Reply:
Yes, since we want to allow multiple queries, this code and the associated exceptions/variables are unnecessary.

Reply:
Now that I've thought about it. It could be a good idea to create an unpause function which either does the deserialization itself or needs to be passed the result of it which also unpauses the graph workers and sets this variable to false. This would enforce that updates cannot happen before the sketches are restored.

Thoughts?

Reply:
I'm not sure what purpose this unpause function will serve. can you elaborate?

Reply:
So basically, our normal code operates with the assumption that CC will only be called once. This new code adds the option to call it again. However, if you insert more updates after CC without resetting the sketches first things will get really weird. 

So this function I'm thinking of would be a graph function which enforces that you must reset the sketches before the graph code will accept another update.

Reply:
It's the function I also discussed in the below comment

Reply:
I see. Let’s do that

Reply:
Okay there's one question I'd like to get yours and Kenny's take on. Should we setup this function to accept the original sketches as a parameter, to do the deserialization itself, or to trust that the user has deserialized before calling it?

Reply:
Oh ... I guess we do have a use for a copy constructor... lol

Reply:
I'm thinking that we should trust the user, since most of the time we'll just want to run the algorithm once and not care about saving the sketches

Reply:
Okay, I wrote up that function. It's in the most recent pull request